packages/agent/src/actor.ts

```diff
@@ -285,11 +312,20 @@ export class Actor {
     interfaceFactory: IDL.InterfaceFactory,
     configuration: ActorConfig,
   ): ActorSubclass<T> {
-    return new (this.createActorClass(interfaceFactory))(
+    return new (this.createActorClass(interfaceFactory, false))(
       configuration,
     ) as unknown as ActorSubclass<T>;
   }
 
+  public static createActorWithHttpDetails<T = Record<string, ActorMethod>>(
+    interfaceFactory: IDL.InterfaceFactory,
+    configuration: ActorConfig,
+  ): ActorSubclass<ActorMethodMappedWithHttpDetails<T>> {
+    return new (this.createActorClass(interfaceFactory, true))(
```

Comment:
It might be worth using an object parameter to simplify adding other configuration options in the future, i.e.:

```suggestion
    return new (this.createActorClass(interfaceFactory, {
      httpDetails: true,
    }))(
```

Reply:
Makes sense, updated it now 👍